internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -392,3 +406,57 @@ func validateRootDirPath(input string) error {
 func validateContainerPath(input string) error {
 	return validatePath(input, maxDockerContainerPathLength)
 }
+
+// ValidatePubSubName validates naming is correct for topics in publishing/subscribing cases, such as naming for a
+// SNS Topic intended for a publisher.
+func validatePubSubName(name *string) error {
+	if name == nil || len(aws.StringValue(name)) == 0 {
+		return errMissingPublishTopicField
+	}
+
+	// name must contain letters, numbers, and can't use special characters besides _ and -
+	if !awsSNSTopicRegexp.MatchString(aws.StringValue(name)) {
+		return errInvalidPubSubTopicName
+	}
+
+	return nil
+}
+
+func validateWorkerNames(names []string) error {
+	for _, name := range names {
+		err := validateName(name)
+		if err != nil {
+			return fmt.Errorf("worker name `%s` is invalid: %w", name, err)
+		}
+	}
+	return nil
+}
+
+func validateName(name string) error {
```

Comment:
It seems like this is only for validating the worker name. Can we rename this or merge its logic within the for loop?

Reply:
Since this function would work for any service name, I renamed it to validateSvcName.